File: backoffice/src/shared/domain/AggregateRoot.py
"""
 *
 * Libraries 
 *
"""

from .DomainEvent import DomainEvent

"""
 *
 * Classes 
 *
"""

class AggregateRoot:

    """
     *
     * Parameters 
     *
    """

    __events : list[ DomainEvent ]

    """
     *
     * Methods 
     *
    """

    def __init__( self ) -> None:
        self.__events = []
    
    def record( self, event : DomainEvent ) -> None:
        self.__events.append( event )
    
    def pullEvents( self ) -> list[ DomainEvent ]:
        # Variables
        events : list[ DomainEvent ]
        # Code
        events = self.__events
        self.__events = []
        return events

File: attention/src/shared/domain/AggregateRoot.cs
/* 
 * 
 * Classes 
 *
*/

namespace attention.src.shared.domain {

    public class AggregateRoot {

        /* 
         * 
         * Attributes
         *
        */

        private List<DomainEvent> _events;

        /* 
         * 
         * Methods
         *
        */

        public AggregateRoot() {
            _events = new List<DomainEvent>();
        }

        public void record( DomainEvent domainEvent ) {
            _events.Add( domainEvent );
        }

        public List<DomainEvent> pullEvents() {
            // Variables
            List<DomainEvent> events;
            // Code
            events  = _events;
            _events = new List<DomainEvent>();
            return events;
        }

    }

}